src/Persistence/YamlValidator/Validator.cs

```diff
@@ -4,40 +4,45 @@
 using Json.Schema;
 using Yaml2JsonNode;
 using System.Text.Json;
+using YamlDotNet.Core;
```

Comment:
in order to isolate dependents from also having to take an active dependency on YamlDotNet and Yaml2JsonNode, etc, be sure that you do not expose any of their types in the public interfaces for this assembly.

For example, we don't want the DocumentServer or Pac CLI to take a dependency on these 3rd party assemblies, as we may change our own dependencies on them over time. We've done this by making sure that we never actually expose any YamlDotNet types in this assembly's public API surface. We create an interface to wrap certain things, and factories etc. Other options is to create our own 'Options' pattern records for capturing the options we want to expose, but use them to configure the YamlDotNet options at runtime.

The same pattern should be done with these new Yaml2JsonNode types.

Where is the `EvaluationOptions` type defined? and the `JsonSchema` type?

One technique to do this would be to create `IValidatorFactory` and `IValidator` interfaces. Then make the `Validator` and `ValidatorFactory` classes have `internal` visibility.

Reply:
> Where is the EvaluationOptions type defined? and the JsonSchema type?
Bot of these types are defined [here](https://github.com/json-everything/json-everything) -> this library is the one used for validation

Reply:
I went with your suggestion of making the `Validator.cs` and `ValidatorFactory.cs` internal as well as making public interfaces for them. In addition to this I applied your suggestion of not exposing the 3rd party assembly types [here](https://github.com/microsoft/PowerApps-Tooling/blob/82f1abbba7bff3e01330015e7af7bf49b55dae23/src/Persistence/YamlValidator/ValidatorError.cs)  in the encapuslated type I made, ValidatorError.cs. Now, instead of accepting a type from the Json.Schema library (EvaluationResults) type it now uses strings, and a dictionary when being instantiated.

Reply:
Another question: since the factory's implementation is internal too wouldn't I need a public provider to instantiate the factory ? More specifically: say I want to use `Microsoft.PowerPlatform.PowerApps.Persistence.YamlValidator;` in PAC cli, instantiating the factory would cause an error. So instead I use a public provider which instantiates and returns a ValidatorFactory interface